[CS2113-W14-4] Pharmacy Inventory & Logistics Ledger (PILL)

yijiano opened the pull request:
Pharmacy Inventory & Logistics Ledger (PILL) is a Command Line Interface (CLI) tool designed to assist in
managing and tracking medicinal inventory.

thomasjlalba commented on src/main/java/seedu/pill/command/HelpCommand.java:
```diff
@@ -0,0 +1,255 @@
+        logger.info("Showing specific help for command: " + command);
+
+        switch (command.toLowerCase()) {
+        case "help":
```
Maybe you can refactor the magic string?

thomasjlalba commented on src/main/java/seedu/pill/util/StringMatcher.java:
```diff
@@ -0,0 +1,58 @@
+                } else if (j == 0) {
+                    dp[i][j] = i;
+                } else {
+                    dp[i][j] = min(dp[i - 1][j - 1]
```
Maybe you can look to refactor this line to make it more readable?

thomasjlalba commented on src/main/java/seedu/pill/util/StringMatcher.java:
```diff
@@ -0,0 +1,58 @@
+        String closestMatch = null;
+        int minDistance = Integer.MAX_VALUE;
+
+        for (String valid : validStrings) {
```
valid might be a bit ambiguous, maybe you can use `validString` instead?

thomasjlalba commented on src/main/java/seedu/pill/util/Storage.java:
```diff
@@ -0,0 +1,157 @@
+    public Item loadLine(String line) throws PillException {
+        Item item;
+        String[] data = line.split(SEPARATOR);
+        if (data.length == 3) {
```
Maybe you can refactor the magic literal to make it more readable

thomasjlalba commented on src/main/java/seedu/pill/util/Storage.java:
```diff
@@ -0,0 +1,157 @@
+        String[] data = line.split(SEPARATOR);
+        if (data.length == 3) {
+            try {
+                item = new Item(data[0], Integer.parseInt(data[1]), LocalDate.parse(data[2]));
```
Maybe you could refactor `data[0]` and so on such that it is more understandable what each variable means before the if statement

thomasjlalba commented on src/main/java/seedu/pill/util/Storage.java:
```diff
@@ -0,0 +1,157 @@
+     * @throws PillException if an error occurs during the saving process.
+     */
+    public void saveItemMap(ItemMap itemMap) throws PillException {
+        try {
+            File file = initializeFile();
+            FileWriter fw = new FileWriter(file);
+
+            for (String itemName : itemMap.items.keySet()) {
+                TreeSet<Item> itemSet = itemMap.items.get(itemName);
+                for (Item item : itemSet) {
+                    fw.write(item.getName() + SEPARATOR + item.getQuantity());
+
+                    if (item.getExpiryDate().isPresent()) {
+                        fw.write(SEPARATOR + item.getExpiryDate().get().toString());
+                    } else {
+                        fw.write(SEPARATOR + "");
+                    }
+
+                    fw.write(System.lineSeparator());
+                }
+            }
```
Maybe you can look at refactoring this as it is deeply nested

thomasjlalba commented on src/main/java/seedu/pill/util/Parser.java:
```diff
@@ -0,0 +1,365 @@
+            quantityStr = splitArguments[quantityIndex];
+            expiryDateStr = null;
+            itemName = buildItemName(splitArguments, 0, quantityIndex);
+        } else if (quantityIndex != null && quantityIndex == splitArguments.length - 2
+                && isValidDate(splitArguments[quantityIndex + 1])) {
```
Maybe you can refactor the conditional statement out to avoid complicated espressions

thomasjlalba commented on src/main/java/seedu/pill/util/Parser.java:
```diff
@@ -0,0 +1,365 @@
+     * @return An {@code AddItemCommand} containing the parsed item name, quantity, and optional expiry date.
+     * @throws PillException If the input format is invalid
+     */
+    private AddItemCommand parseAddItemCommand(String arguments) throws PillException {
```
Maybe you can split this method up to multiple smaller methods instead of it all being in this long method

thomasjlalba commented on src/main/java/seedu/pill/util/ItemMap.java:
```diff
@@ -0,0 +1,335 @@
+                    item.setQuantity(newQuantity);
+                    itemUpdated = true;
+                    expiryDate.ifPresentOrElse(
+                            expiry -> {
+                                LOGGER.info("Updated existing item with expiry date: " + newItem);
+                                System.out.println("Item already exists with the same expiry date. Updated quantity: \n"
+                                        + newItem);
```
Maybe you can avoid lambda expressions to make it more readable for readers

thomasjlalba commented on src/main/java/seedu/pill/util/ItemMap.java:
```diff
@@ -0,0 +1,335 @@
+                if (itemSet.isEmpty()) {
+                    items.remove(name);
+                }
+            } else {
+                LOGGER.warning("Attempt to delete non-existent item: " + removedItem);
+                System.out.println("Item not found: " + removedItem);
+            }
+        } else {
+            LOGGER.warning("Attempt to delete non-existent item: " + name);
+            System.out.println("Item not found: " + name);
+        }
```
Maybe you can make the [happy path more obvious](https://nus-cs2113-ay2425s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent) by switching this into guard clauses instead?

thomasjlalba commented on src/main/java/seedu/pill/util/ItemMap.java:
```diff
@@ -0,0 +1,335 @@
+     *
+     * @param newItem The item to be added.
+     */
+    public void addItem(Item newItem) {
```
Maybe can look into splitting this method into several smaller methods as it is currently quite long

thomasjlalba commented on src/main/java/seedu/pill/util/ItemMap.java:
```diff
@@ -0,0 +1,335 @@
+                throw new PillException(ExceptionMessages.INVALID_QUANTITY);
+            }
+
+            List<Item> filteredItems = items.values().stream()
+                    .flatMap(TreeSet::stream)
+                    .filter(item -> item.getQuantity() <= threshold)
+                    .toList();
```
Maybe you can look into refactoring this to make it more readable and less complicated?

thomasjlalba commented on src/main/java/seedu/pill/util/Parser.java:
```diff
@@ -0,0 +1,365 @@
+            String arguments = String.join(" ", Arrays.copyOfRange(splitInput, 1, splitInput.length));
+
+            switch (commandString) {
+            case "exit":
```
Maybe you can look into refactoring these magic strings?

thomasjlalba commented on src/main/java/seedu/pill/command/DeleteItemCommand.java:
```diff
@@ -0,0 +1,36 @@
+
+    @Override
+    public void execute(ItemMap itemMap, Storage storage) throws PillException {
+        // Looks stupid, but this way I don't handle Optionals in this class
```
Maybe you can rephrase this to [write comments targeting other programmers reading the code](https://nus-cs2113-ay2425s1.github.io/website/se-book-adapted/chapters/codeQuality.html#write-to-the-reader)

thomasjlalba commented on src/main/java/seedu/pill/command/HelpCommand.java:
```diff
@@ -0,0 +1,255 @@
+            if (closestMatch != null) {
+                System.out.println("Did you mean: " + closestMatch + "?");
+                showSpecificHelp(closestMatch);
+            } else {
+                suggestSimilarCommand(command);
+            }
```
Maybe this can be changed to a guard clause instead to make the happy path more prominent

thomasjlalba commented on src/main/java/seedu/pill/command/HelpCommand.java:
```diff
@@ -0,0 +1,255 @@
+    private final String commandName;
+    private final boolean verbose;
+
+    public HelpCommand(String commandInput, boolean verbose) {
```
Maybe you can change the boolean variable name to make it more obvious it is a boolean

thomasjlalba reviewed (commented):
Do remember to ensure your code is more readable for others and to simplify expressions. Keep up the good work

Bev-low commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,252 @@
+
+<!-- @@author yakultbottle -->
+
+### Storage
```
I think these classes can have UML diagrams

Bev-low commented on docs/DeveloperGuide.md:
````diff
@@ -1,38 +1,252 @@
+Example:
+
+```
+AddItemCommand command = new AddItemCommand(itemName, quantity, expiryDate);
````
I think this can be put in the implementation section, following the example they give us.

Bev-low commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,252 @@
 
+**API**: PillLogger.java
+
+<img src = "diagrams/PillLogger.png" alt="Component Diagram for PillLogger"/>
```
This diagram looks like it doesnt follow the standard

YukeeHong replied:
yeah, i think missing some tag and some place should use dotted lines

Bev-low commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,252 @@
+The ItemMap class contains a key-value pair, implemented as a Map, from the
+item name(String) to the item(TreeSet\<Item>)
+
+![](diagrams/ItemMap-ClassDiagram.png)
```
I dont think this follows the standard also, I am not sure what the dotted box "item" mean on comparable

Bev-low replied:
should do comparable:item

Bev-low reviewed (commented):
Overall looks neat

YukeeHong commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,252 @@
+
+The Item class has three private variables, a name, a quantity, and an
+expiry date. An Item may or may not have an expiry date, so we store it
+as an Optional, which handles empty values for us without using null. 
```
could briefly explain why Optional was chosen over null

YukeeHong commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,252 @@
+* Technical Requirements: Any mainstream OS, i.e. Windows, macOS or Linux, with Java 11 installed. Instructions for downloading Java 11 can be found [here](https://www.oracle.com/sg/java/technologies/javase/jdk11-archive-downloads.html).
+* Project Scope Constraints: The application should only be used for tracking. It is not meant to be involved in any form of monetary transaction.
+* Project Scope Constraints: Data storage is only to be performed locally.
+* Quality Requirements: The application should be able to be used effectively by a novice with little experience with CLIs.
```
can be more detailed and concrete

YukeeHong commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,252 @@
+
+#### AddItemCommand
+
+<img src = "diagrams/AddItemCommand-SequenceDiagram.png"/>
```
some part like print  and log can be simplified

DanLinhHuynh-Niwashi replied:
Would it be better to add some activation bars to show if the objects are working?

LTK-1606 replied:
Are these notations correct? Should it include activation bars?

DanLinhHuynh-Niwashi commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,252 @@
+
+The high-level overview of the project structure is as follows:
+
+<img src = "diagrams/High-Level-Overview.png" alt="High Level Overview of PILL"/>
```
You may try to consider the direction of arrows. Since it's is MVC model, the arrow from View to Controller might be a two way arrow (e.g. View send request to Controller and Controller response with data?), the same for Controller to Model components

DanLinhHuynh-Niwashi commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,252 @@
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The project is designed using the Model-View-Controller (MVC) architecture, with the following components:
```
Since Model - Controller - View is at a higher level comparing to other classes (e.g. Model contains Item and ItemClass) and they are not really a class as well, you may try to create a hierarchy, just to distinguish between the architecture and the actual implementation. For example:
3. Controller:
3.1. Item
3.2. ItemMap

DanLinhHuynh-Niwashi commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,252 @@
+
+The UI consists of the following components:
+
+1. **Parser**: The Parser class is responsible for interpreting user input and returning the corresponding command.
```
As I think UI is the 'View' part in your architecture, should it be better to make this parser a part of the Controller instead of View? As the Controller in MVC is responsible for handling user input, processing it, and coordinating between the Model and the View, I think the parser is nearer to a Controller component than a View component

DanLinhHuynh-Niwashi commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,252 @@
+
+<!-- @@author cxc0418 -->
+
+### UI and I/O
```
Will it be good to put these implementation into a hierarchy corresponding to your architecture? To make a clearer view of how each component related to the high-level architecture!

LTK-1606 replied:
Should there be more visuals and diagrams to make this clearer?

Zackermax commented on docs/diagrams/High-Level-Overview.puml:
The diagram should not have class symbols (green C) and the arrows should be orthogonal.

Zackermax commented on docs/diagrams/AddItemCommand-SequenceDiagram.puml:
The diagram is missing activation boxes

Zackermax commented on docs/diagrams/PillLogger.puml:
The arrows should be orthogonal.

LTK-1606 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,252 @@
+expiry date. An Item may or may not have an expiry date, so we store it
+as an Optional, which handles empty values for us without using null. 
+
+![](diagrams/Item-ClassDiagram.png)
```
Are your notations used correctly? Should it be "function(): 'return type'"?

Zackermax commented on docs/diagrams/AddItemCommand-SequenceDiagram.puml:
The formatting for the "else" condition in the second section is not formatted properly.

LTK-1606 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,252 @@
+        - [Commands](#commands)
+        - [Storage](#storage)
+        - [Item and ItemMap](#item-and-itemmap)
+        - [DateTime](#datetime)
+        - [Exceptions and Logging](#exceptions-and-logging)
```
Are these links supposed to be here? Should they be broken?

LTK-1606 reviewed (commented):
Looks neat, but notations used may not follow requirements, some broken links and maybe more diagrams would be better

cxc0418 committed:
Merge branch 'master' of https://github.com/cxc0418/tp

yakultbottle committed:
Add execute method for UseItemCommand

cxc0418 committed:
Update on UG

cxc0418 committed:
Update on UG

yakultbottle committed:
Update execute method to follow convention in other ItemCommands

cxc0418 committed:
Update on UG

yakultbottle committed:
Add test coverage for UseItemCommand

yakultbottle committed:
Correct checkstyle errors

yakultbottle committed:
Merge pull request #135 from yakultbottle/priorityRemoval

Implement Use command with priority removal

cxc0418 committed:
Merge branch 'master' into master

cxc0418 committed:
Resolve conflict

cxc0418 committed:
Resolve conflict

cxc0418 committed:
resolve conflict

cxc0418 committed:
Merge pull request #134 from cxc0418/master

Update on the cost & price etc

philip1304 committed:
Comments // Updated

Notes:
- Added in-line comments for clarity.

philip1304 committed:
TimestampIO // Implemented

Notes:
- Added TimestampIO.java

philip1304 committed:
JUnit Tests // Updated

Notes:
- Added new test cases for coverage in HelpCommandTest.java

philip1304 committed:
TimestampIOTest // Implemented

Notes:
- Added TimestampIOTest.java

philip1304 committed:
Indentation // Updated

philip1304 committed:
Variable Declaration // Fixed

philip1304 committed:
TimestampIO & TimestampIOTest // Updated

philip1304 committed:
no message

philip1304 committed:
ASDFPLSPLSPLSPLSPLS

philip1304 committed:
Merge pull request #136 from philip1304/master

Timestamps, JUnit Tests, Comments // Implemented

yakultbottle committed:
Fix incorrect Item and ItemMap class diagrams

yakultbottle committed:
Merge pull request #137 from yakultbottle/fixClassDiagrams

Fix incorrect Item and ItemMap class diagrams

yijiano committed:
Update High Level Overview Diagram

yijiano committed:
More documentation changes

yijiano committed:
Add PPP

yijiano committed:
Add empty line at end of PPP

yijiano committed:
Fix format

yijiano committed:
Update AddItem Diagram

yijiano committed:
Merge pull request #139 from yijiano/fix/docs

Update DG, add PPP

yijiano committed:
Fix PPP

yijiano committed:
Merge pull request #140 from yijiano/fix/PPP

Fix PPP

cnivedit committed:
Create PPP

cnivedit committed:
Merge branch 'master' into ppp/cnivedit

cnivedit committed:
Edit contribution list order based on review comment

cnivedit committed:
Merge pull request #141 from cnivedit/ppp/cnivedit

Draft Project Portfolio Page

cnivedit committed:
Create OrderCommand class

cnivedit committed:
Create TransactionsCommand class

cnivedit committed:
Create ViewOrdersCommand class

cnivedit committed:
Create FulfillCommand class

cnivedit committed:
Create TransactionHistoryCommand class

cnivedit committed:
Instantiate TransactionManager

cnivedit committed:
Add exception message for transaction error

cnivedit committed:
Update data structures

cnivedit committed:
Fix wrong ItemMap instance passing

cnivedit committed:
Update exception messages

cnivedit committed:
Create method to list items

cnivedit committed:
Edit to use Item and ItemMap

cnivedit committed:
Create methods to list orders and transactions

cnivedit committed:
Integrate new commands

cnivedit committed:
Fix CI failing

cnivedit committed:
Fix JavaDoc exceeding line count

cnivedit committed:
Fix incorrect JavaDoc line indentation

cnivedit committed:
Update indentation

cnivedit committed:
Remove deprecated test with regards to item map

cnivedit committed:
Update tests with latest changes

cnivedit committed:
Remove unused imports

cnivedit committed:
Shorten JavaDoc based on review comment

cnivedit committed:
Extract helper function based on review comment

cnivedit committed:
Fix line count and indentation

cnivedit committed:
Merge pull request #142 from cnivedit/dev/order-transaction-management

Manage orders and transactions

cnivedit committed:
Print order details upon creation

cnivedit committed:
Create getRawInput method

cnivedit committed:
Merge pull request #160 from cnivedit/bugfix/order-placed-message

Print order details upon creation

cnivedit committed:
Merge branch 'master' into bugfix/excessive-newlines

cnivedit committed:
Merge pull request #161 from cnivedit/bugfix/excessive-newlines

Fix excessive newline printing

cnivedit committed:
Add no transactions message

yakultbottle committed:
Create Exceptions and Logging

cnivedit committed:
Merge pull request #162 from cnivedit/bugfix/no-transactions-message

Add no transactions message

yakultbottle committed:
Remove redundant logging portion

yakultbottle committed:
Merge branch 'master' into DG

yakultbottle committed:
Merge pull request #164 from yakultbottle/DG

Update DG with Exceptions

cnivedit committed:
Edit stock-check message

cnivedit committed:
Fix line count

cxc0418 committed:
Updated  the toString method to ensure all price/cost values are printed to two decimal places using %.2f.
Updated SetCostCommand, SetPriceCommand, and isANumber method to support double values and format them correctly to %.2f.
Optimized the saveItemMap method to save cost and price attributes formatted as %.2f.
Updated PPP.

cxc0418 committed:
Merge branch 'AY2425S1-CS2113-W14-4:master' into master

cxc0418 committed:
Merge branch 'AY2425S1-CS2113-W14-4:master' into master

cnivedit committed:
Update tests with new stock-check message

cnivedit committed:
Merge pull request #167 from cnivedit/bugfix/stock-check-message

Edit stock-check message

cxc0418 committed:
Merge branch 'master' of https://github.com/cxc0418/tp

cxc0418 committed:
Merge branch 'master' of https://github.com/cxc0418/tp

cxc0418 committed:
Merge branch 'master' of https://github.com/cxc0418/tp

cxc0418 committed:
Merge branch 'master' of https://github.com/cxc0418/tp

cxc0418 committed:
Merge branch 'master' of https://github.com/cxc0418/tp

cxc0418 committed:
Merge branch 'master' into master

cnivedit committed:
Fix updated quantity bug

cxc0418 committed:
Merge pull request #168 from cxc0418/master

Improve cost/price command functionality

cnivedit committed:
Merge branch 'master' into bugfix/additem-message

cnivedit committed:
Merge pull request #169 from cnivedit/bugfix/additem-message

Fix updated quantity bug

yijiano committed:
Fix Help, UG

yijiano committed:
Merge branch 'master' into fix/UG

yijiano committed:
Fix CI errors

yijiano committed:
Merge branch 'fix/UG' of https://github.com/yijiano/tp into fix/UG

yijiano committed:
Fix README.md

yijiano committed:
Fix HelpCommand test cases

yijiano committed:
Add newline to end of HelpCommandTest

yijiano committed:
Merge pull request #171 from yijiano/fix/UG

Fix Help, UG

yakultbottle committed:
Add Parser and Order

yakultbottle committed:
Add Parser and Order

yakultbottle committed:
Merge branch 'master' into DG-Parser

yijiano committed:
Fix README.md and some website features

yakultbottle committed:
Merge pull request #172 from yakultbottle/DG-Parser

Add Parser and Order to DG

yijiano committed:
Merge pull request #173 from yijiano/fix/AboutUs

Fix README.md and some website features

cxc0418 committed:
Closes issue 204, 201, 200, 197, 196, 192, 190, 181, 186, 180, 179, 166, 216, 176, 177

cxc0418 committed:
Closes issue 204, 201, 200, 197, 196, 192, 190, 181, 186, 180, 179, 166, 216, 176, 177

cxc0418 committed:
Merge branch 'master' of https://github.com/cxc0418/tp

cxc0418 committed:
Merge branch 'master' of https://github.com/cxc0418/tp

cxc0418 committed:
Merge branch 'master' of https://github.com/cxc0418/tp

cnivedit committed:
Fix inaccurate order command format

cxc0418 committed:
Merge pull request #223 from cxc0418/master

Closes issues 204, 201, 200, 197, 196, 192, 190, 181, 186, 180, 179, 1…

cnivedit committed:
Merge pull request #224 from cnivedit/bugfix/ug-ordercommand

Fix inaccurate order command format

cxc0418 committed:
Implemented visualizer class and commands for data visualization (price, cost, stock, and cost-price comparison)
Updated PPP, UG and DG
Closed issues 222, 205, 166, 174, 226, 221,

cxc0418 committed:
Implemented visualizer class and commands for data visualization (price, cost, stock, and cost-price comparison)
Updated PPP, UG and DG
Closed issues 222, 205, 166, 174, 226, 221,

cxc0418 committed:
Merge remote-tracking branch 'origin/master'

cxc0418 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/seedu/pill/command/VisualizeCostCommand.java
#	src/main/java/seedu/pill/command/VisualizeCostPriceCommand.java
#	src/main/java/seedu/pill/command/VisualizePriceCommand.java
#	src/main/java/seedu/pill/command/VisualizeStockCommand.java
#	src/main/java/seedu/pill/util/Visualizer.java

cxc0418 committed:
Merge branch 'master' of https://github.com/cxc0418/tp

cxc0418 committed:
more update on UG

cxc0418 committed:
Merge pull request #227 from cxc0418/master

Implemented visualizer class and commands for data visualization (pri…

yakultbottle committed:
Separate AddItemCommand Sequence diagram

yakultbottle committed:
Update DG with new diagram

yakultbottle committed:
Merge branch 'master' into sequence-diagram-edit

yakultbottle committed:
Merge pull request #229 from yakultbottle/sequence-diagram-edit

Shorten AddItemCommand Sequence Diagram

yakultbottle committed:
Fix incorrect TreeSet instantiation

yakultbottle committed:
Update colours for better clarity

yakultbottle committed:
Merge pull request #230 from yakultbottle/sequence-fix

Fix incorrect TreeSet instantiation in AddItemCommand sequence diagram

yijiano committed:
Fix #207

yijiano committed:
Improve value proposition

yijiano committed:
Add transaction commands to UG, add no special characters rule

yijiano committed:
Change storage data file type

yijiano committed:
Fix use, and edit descriptions

yijiano committed:
Stopped trolling

yakultbottle committed:
Remove circular icons in class diagram

yakultbottle committed:
Update Parser section in DG

yakultbottle committed:
Update Glossary in DG

yijiano committed:
Merge pull request #231 from yijiano/fix/UGAgain

Fix UG-related issues from PED

yakultbottle committed:
Merge branch 'master' into command-sequence-diagrams

yijiano committed:
Fix add command case sensitivity

yakultbottle committed:
Add more terms to Glossary

yakultbottle committed:
Merge pull request #232 from yakultbottle/command-sequence-diagrams

yijiano committed:
Update README.md

yakultbottle committed:
Rearrange User Guide to put Notes on top

cnivedit committed:
Update fulfill order to use UUID in place of index

cnivedit committed:
Display message if no orders found when using view-orders command

cnivedit committed:
Update UG to use order UUID

cnivedit committed:
Update help message for fulfill-order

cnivedit committed:
Merge branch 'master' into bugfix/noorder-message

cnivedit committed:
Merge branch 'master' into bugfix/fulfillargs

cnivedit committed:
Merge pull request #236 from cnivedit/bugfix/noorder-message

Display message if no orders found when using view-orders command

cnivedit committed:
Merge pull request #235 from cnivedit/bugfix/fulfillargs

Update fulfill order to use UUID in place of index

yakultbottle committed:
Spring cleaning of unused files

yakultbottle committed:
Merge branch 'master' into user-guide

yakultbottle committed:
Merge pull request #234 from yakultbottle/user-guide

Rearrange User Guide to put Important Notes on top

cnivedit committed:
Update help message for order command

cnivedit committed:
Merge branch 'master' into bugfix/order-help

cnivedit committed:
Merge pull request #237 from cnivedit/bugfix/order-help

Update help message for order command

yijiano committed:
Add a whole bunch of test cases

cnivedit committed:
Edit to use date range for transaction history

yijiano committed:
Purge all unused imports

yijiano committed:
Merge branch 'master' into test/MoreTestCases

yakultbottle committed:
Add notes to OrderCommand

yijiano committed:
Clean post merge

yijiano committed:
Merge branch 'test/MoreTestCases' of https://github.com/yijiano/tp into test/MoreTestCases

cnivedit committed:
Update tests

yijiano committed:
Fix more test case issues

cnivedit committed:
Remove unused imports

yijiano committed:
Update PPP

yijiano committed:
Remove more unused dependencies

cnivedit committed:
Merge pull request #238 from cnivedit/dev/transaction-history

Edit to use date range instead of time range for transaction history

yijiano committed:
Merge branch 'master' into test/MoreTestCases

yijiano committed:
Ressolve more test errors

yijiano committed:
Appreciate these CI errors

yakultbottle committed:
Add expiryDate field for createTransaction

yijiano committed:
Update UG

yijiano committed:
Fix Lower Case Type Fixing

yijiano committed:
Fix errors

yijiano committed:
Merge pull request #233 from yijiano/test/MoreTestCases

Fix #183, Test Cases

yakultbottle committed:
Merge branch 'master' into order-transaction-bugs

yakultbottle committed:
Define expiryDate behaviour for dispense orders

yakultbottle committed:
Delete extra method that came from nowhere

yakultbottle committed:
Merge pull request #240 from yakultbottle/order-transaction-bugs

Fix Order and Transaction bugs

cnivedit committed:
Elaborate on transactions and order

cnivedit committed:
Merge branch 'master' into devguide/transaction-management

yakultbottle committed:
fix CSV loading

cnivedit committed:
Update sections, edit table of contents

cnivedit committed:
Merge branch 'devguide/transaction-management' of https://github.com/cnivedit/tp into devguide/transaction-management

cnivedit committed:
Merge pull request #241 from cnivedit/devguide/transaction-management

Elaborate on transactions and order

philip1304 committed:
Updated JavaDoc & JUnit Tests

philip1304 committed:
Merge branch 'master' of https://github.com/philip1304/tp

yakultbottle committed:
Implement save to storage after fulfilling order

yakultbottle committed:
Merge branch 'master' into csv-fix

philip1304 committed:
Fixes

philip1304 committed:
Fixes

philip1304 committed:
Fixes

philip1304 committed:
Fixes

yakultbottle committed:
Merge pull request #243 from yakultbottle/csv-fix

Escape CSV characters and save fulfilled orders to storage

cnivedit committed:
Update items section

cnivedit committed:
Merge branch 'master' into devguide/items

cnivedit committed:
Merge pull request #245 from cnivedit/devguide/items

Update items section

cnivedit committed:
Update dependency and interface relationships

cnivedit committed:
Merge branch 'master' into devguide/logger-componentdiagram

cnivedit committed:
Merge pull request #246 from cnivedit/devguide/logger-componentdiagram

Update dependency and interface relationships

philip1304 committed:
Merge branch 'master' into master

philip1304 committed:
Fixes

cnivedit committed:
Update PPP

philip1304 committed:
PPP

philip1304 committed:
Merge branch 'master' of https://github.com/philip1304/tp

yakultbottle committed:
Update PPP

cnivedit committed:
Add new sections

cnivedit committed:
Merge pull request #247 from cnivedit/ppp/cnivedit-1

Update PPP

yakultbottle committed:
Merge branch 'master' into final-ppp

yakultbottle committed:
Merge pull request #248 from yakultbottle/final-ppp

Update PPP

philip1304 committed:
Merge pull request #244 from philip1304/master

JUnit Tests

cnivedit committed:
Update high level architecture diagram

cnivedit committed:
Update to include other util classes

cnivedit committed:
Fix reposense link

cnivedit committed:
Merge pull request #251 from cnivedit/bugfix/ppp

Fix reposense link

philip1304 committed:
ppppppp

philip1304 committed:
Merge branch 'AY2425S1-CS2113-W14-4:master' into master

cnivedit committed:
Fix formatting

cnivedit committed:
Fix spacing

cnivedit committed:
Merge branch 'master' into bugfix/ppp-formatting

cnivedit committed:
Update spacing

philip1304 committed:
Merge pull request #252 from philip1304/master

PPP

cnivedit committed:
Merge pull request #253 from cnivedit/bugfix/ppp-formatting

Fix PPP formatting

yijiano committed:
Merge pull request #249 from cnivedit/devguide/overview

Update high level architecture diagram

yijiano committed:
Fix DG

yijiano committed:
Fix UG, DG diagram

yijiano committed:
Fix format

yijiano committed:
Merge pull request #255 from yijiano/fix/DG-UG

Fix DG and UG

yakultbottle committed:
Fix UG formatting

yakultbottle committed:
Merge pull request #256 from yakultbottle/master

yakultbottle committed:
UG and DG small fixes

yakultbottle committed:
Merge pull request #257 from yakultbottle/docs-triage

UG and DG small fixes

yakultbottle committed:
minor change to PPP

yakultbottle committed:
Merge pull request #258 from yakultbottle/final-PPP

minor change to PPP

cnivedit committed:
Update PPP

cnivedit committed:
Merge branch 'master' into ppp/cnivedit-2

cnivedit committed:
Merge pull request #259 from cnivedit/ppp/cnivedit-2

Update PPP

cnivedit committed:
Add page break and image widths

yakultbottle committed:
Add acknowledgement for XChart

cnivedit committed:
Merge branch 'master' into ppp/cnivedit-3

cnivedit committed:
Merge pull request #261 from cnivedit/ppp/cnivedit-3

Update PPP formatting

yakultbottle committed:
Hopefully fix visualize

yakultbottle committed:
Merge branch 'master' into ack

yakultbottle committed:
common checkstyle L

yakultbottle committed:
Merge pull request #260 from yakultbottle/ack

Add acknowledgement for XChart

yijiano committed:
Update AboutUs.md
